beacon-chain/blockchain/chain_info.go

```diff
@@ -332,7 +334,14 @@ func (s *Service) IsOptimisticForRoot(ctx context.Context, root [32]byte) (bool,
 		return false, err
 	}
 	if ss == nil {
-		return false, errInvalidNilSummary
+		// if the requested root is the headroot we should treat the
+		// node as optimistic. This can happen if we pruned INVALID
+		// nodes and no viable head is available.
+		headRoot, err := s.HeadRoot(ctx)
+		if err == nil && bytes.Equal(headRoot, root[:]) {
+			return true, nil
+		}
+		return true, errInvalidNilSummary
```

Comment:
We kinda lose info if `err != nil`. How about?

```suggestion
                 if err != nil {
                         return true, err
                 }
		if bytes.Equal(headRoot, root[:]) {
			return true, nil
		}
		return true, errInvalidNilSummary
```

Reply:
Would it be helpful to add a metric counter for `bytes.Equal(headRoot, root[:])` ?